### PR TITLE
Basic linter config

### LINKED ISSUE #246: Set up linting pre-commit hook
## Expected Behavior

When someone opens a pull request, their changes should be automatically linted before code review.

## Actual Behavior

Currently, changes must be  linted manually, and there are no automated checks that prevent introducing code that goes against our style guide. 

## Proposed Solution
Set up a linting pre-commit hook such as the one in [tfx-addons](https://github.com/tensorflow/tfx-addons/tree/main/.github). Add documentation on how to lint changes to the contributing guide or create an issue for this.

### PR DESCRIPTION
**Pull request by hanneshapke**
# What does this pull request do?

This PR set up a basic linter configuration for the MCT repo. It is a replacement for the closed PR: https://github.com/tensorflow/model-card-toolkit/pull/242

The `pylintrc` is mimic'ing Google's `pylintrc`: https://github.com/google/styleguide/blob/gh-pages/pylintrc
with two additional disabled checks (due to quantity):
*  abstract-class-instantiated
* protected-access

All other initial issues have been addressed to make the code base compliant with the `pylintrc`:
* unused imports
* super-with-arguments
* inconsistent-quotes
* unneeded not
* line-too_long

Fixes #246

## How did you test this change?

No functionality of tests changes - only formatting and styling of the code itself.

## How did you document this change?

`CONTRIBUTING.md` has been updated

## Before submitting

Before submitting a pull request, please be sure to do the following:
- [x] Read the [How to Contribute guide](https://github.com/tensorflow/model-card-toolkit/blob/master/CONTRIBUTING.md) if this is your first contribution.
- [x] Open an issue or discussion topic to discuss this change.
- [x] Write new tests if applicable.
- [x] Update documentation if applicable.

### CODE REVIEW

**Comment by hanneshapke** [comment]
Thank you, @codesue. Get well soon!

File: model_card_toolkit/utils/validation_test.py [hunk 1]
```diff
@@ -16,13 +16,15 @@
...
-from absl.testing import absltest
-from absl.testing import parameterized
+
 import jsonschema
+from absl.testing import absltest, parameterized
+
 from model_card_toolkit.utils import validation
 
 _MODEL_DETAILS_V1_DICT = {
-    "name": "my model",
```
Comment (codesue) [inline model_card_toolkit/utils/validation_test.py:25]: Does linting fail if this is reverted?

File: tools/build_api_docs.py [hunk 2]
```diff
@@ -2,30 +2,25 @@
...
-from tensorflow_docs.api_generator import generate_lib
-from tensorflow_docs.api_generator import public_api
-
 FLAGS = flags.FLAGS
 
-flags.DEFINE_string(
-    "output_dir",
-    default="/tmp/model_card_toolkit",
-    help="Where to output the docs")
+flags.DEFINE_string("output_dir",
```
Comment (codesue) [inline tools/build_api_docs.py:12]: I personally find the original way of putting the arguments on a new line easier to read. I'm not familiar enough with yapf to know exactly which rule controls this. Maybe SPLIT_BEFORE_EXPRESSION_AFTER_OPENING_PAREN or SPLIT_BEFORE_FIRST_ARGUMENT?

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+name: Lint
```
Comment (codesue) [inline .github/workflows/lint.yml:1]: Please link to a successful run of this workflow.

**Comment by hanneshapke** [comment]
Passing linter run: https://github.com/tensorflow/model-card-toolkit/actions/runs/4520740897
Updated the line breaks after parens

**Review by codesue — APPROVED** [review]

**Pull request merged**